Sources/PortalSwift/Core/Api/PortalYieldXyzApi.swift

```diff
@@ -0,0 +1,309 @@
+//
+//  PortalYieldXyzApi.swift
+//  PortalSwift
+//
+//  Created by Ahmed Ragab on 20/10/2025.
+//
+
+import Foundation
+
+/// API class specifically for Yield.xyz integration functionality.
+///
+/// This class handles all yield-related API calls including discovering yields,
+/// entering/exiting yield opportunities, managing yields, and tracking transactions.
+public class PortalYieldXyzApi {
+    private let apiKey: String
+    private let baseUrl: String
+    private let requests: PortalRequestsProtocol
+    private let logger = PortalLogger()
+    
+    /// Create an instance of PortalYieldXyzApi.
+    /// - Parameters:
+    ///   - apiKey: The Client API key.
+    ///   - apiHost: The Portal API hostname.
+    ///   - requests: An instance of PortalRequestsProtocol to handle HTTP requests.
+    public init(
+        apiKey: String,
+        apiHost: String = "api.portalhq.io",
+        requests: PortalRequestsProtocol? = nil
+    ) {
+        self.apiKey = apiKey
+        self.baseUrl = apiHost.starts(with: "localhost") ? "http://\(apiHost)" : "https://\(apiHost)"
+        self.requests = requests ?? PortalRequests()
+    }
+    
+    /*******************************************
+     * Public functions
+     *******************************************/
+    
+    /// Retrieves yield opportunities from the Yield.xyz integration.
+    /// - Parameter request: The parameters for the yield discovery request.
+    /// - Returns: A `GetYieldsXyzResponse` containing available yield opportunities.
+    /// - Throws: An error if the operation fails.
+    public func getYields(request: GetYieldsXyzRequest) async throws -> GetYieldsXyzResponse {
+        var queryParams: [String] = []
+        
+        // Helper function to add query parameters
+        func addParam(_ key: String, _ value: Any?) {
```

Comment:
We define this function a few times, perhaps it would make sense to put in our request class or even as a private method on this class?